src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -28,7 +28,10 @@ public override void WriteCodeElement(CodeMethod codeElement, LanguageWriter wri
             var queryStringParam = codeElement.Parameters.OfKind(CodeParameterKind.QueryParameter);
             var headersParam = codeElement.Parameters.OfKind(CodeParameterKind.Headers);
             foreach(var parameter in codeElement.Parameters.Where(x => !x.Optional).OrderBy(x => x.Name)) {
-                writer.WriteLine($"_ = {parameter.Name} ?? throw new ArgumentNullException(nameof({parameter.Name}));");
+                if(nameof(String).Equals(parameter.Type.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
The logics for defensive programing (this) and for properties assignement from parameters are at two very different places.
As Michael mentionned the code is already complex and I don't want to add a bunch of additional conditions here.
This is already a shorthand for `if(httpCore == null) throw new ArgumentNullException(nameof(httpCore));`
Now, if you feel this notation style is more confusing than anything, I'm happy to revert it to the more conventional style instead. This would avoid leading people into thinking we could save an additional line). Thoughts?

Reply:
I understand that line of reasoning and I am happy to leave it as it is.